src/views/Detail.vue

```diff
@@ -75,52 +76,63 @@
           v-for="(item, index) in items"
           :key="index"
         >
-          <ImageDetail
-            v-if="fileType === 'images'"
-            :file="item"
-          />
-          <DirectoryDetail
-            v-if="fileType === 'directories'"
-            :file="item"
-          />
-          <DocumentDetail
-            v-if="fileType === 'text'"
-            :file="item"
-          />
-          <AudioDetail
-            v-if="fileType === 'audio'"
-            :file="item"
-          />
-          <VideoDetail
-            v-if="fileType === 'video'"
+          <component
+            :is="componentType"
             :file="item"
           />
         </v-carousel-item>
       </v-carousel>
+      <component
+        v-else
+        :is="componentType"
+        :file="singleItem"
+      />
     </div>
   </div>
 </template>
 
 <script>
-import AudioDetail from '@/components/results/detail/AudioDetail';
-import DocumentDetail from '@/components/results/detail/DocumentDetail';
-import VideoDetail from '@/components/results/detail/VideoDetail';
-import DirectoryDetail from '@/components/results/detail/DirectoryDetail';
-import ImageDetail from '@/components/results/detail/ImageDetail';
 import store from '@/store';
+import { Types } from '../helpers/typeHelper';
+import ImageDetail from '../components/results/detail/ImageDetail';
+import DocumentDetail from '../components/results/detail/DocumentDetail';
+import DirectoryDetail from '../components/results/detail/DirectoryDetail';
+import VideoDetail from '../components/results/detail/VideoDetail';
+import AudioDetail from '../components/results/detail/AudioDetail';
+import { apiMetadataQuery, pageSize } from '../helpers/ApiHelper';
```

Comment:
I don't think that going to any API without going through the store is a good idea. Rather, I would suggest being consistent in the abstraction: `Component <-> Store <-> Search/Metadata API`. This makes it clear how asynchronous state changes work, which could make our lives a *lot* easier once we change the API and/or move towards more complicated workflows.

I'm ok merging this as-is, but would suggest creating an issue with a specific reference to this line number so we can follow up.

Reply:
That would mean setting up a new store module to store deeplinked detail page or something along those lines. I think in this case it adds more complexity than it is worth.

Reply:
I think towards the future, this makes sense. For now, the current approach is good enough.